### PR TITLE
Add reduced parameter grids

### PR DESCRIPTION
**Pull request by gwarmstrong**
I have added or updated the reduced parameter grids for the following regression algorithms:

* RandomForestRegressor
* KNeighborsRegressor
* ElasticNet
* Lasso
* Ridge
* Linear Regression
* MLPRegressor

And the following classification algorithms:
* KNeighborsClassifier
* LogisticRegression
* MLPClassifier
* RandomForestClassifier
* RidgeClassifier

### CODE REVIEW

**Comment by gwarmstrong** [comment]
@patrickimran

File: q2_mlab/tests/test_parameters.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+class ParameterGridsTests(unittest.TestCase):
+
+    regression_algorithms = [
+        "ElasticNet",
+        "KNeighborsRegressor",
+        "Lasso",
+        "LinearRegression",
+        "MLPRegressor",
+        "RandomForestRegressor",
+        "RidgeRegressor",
```
Comment (gwarmstrong) [inline q2_mlab/tests/test_parameters.py]: @patrickimran the grids you added should be added to these class variables so those grids can be tested.

File: .travis.yml [hunk 2]
```diff
@@ -11,7 +11,7 @@ before_install:
   - conda update -q conda
   - conda info -a
 install:
-  - wget -q https://raw.githubusercontent.com/qiime2/environment-files/master/latest/staging/qiime2-latest-py36-linux-conda.yml
+  - wget -q https://raw.githubusercontent.com/qiime2/environment-files/master/2020.6/release/qiime2-2020.6-py36-linux-conda.yml
   - conda env create -q -n test-env --file qiime2-latest-py36-linux-conda.yml
```
Comment (gwarmstrong) [inline .travis.yml]:
```suggestion
  - conda env create -q -n test-env --file qiime2-2020.6-py36-linux-conda.yml
```

Comment (patrickimran) [reply .travis.yml]: good catch

File: q2_mlab/learningtask.py [hunk 3]
```diff
@@ -45,6 +47,21 @@
...
+class ModelFactory:
+    @staticmethod
+    def get_model(model_class, **outer_kwargs):
+        class SpecificModel(model_class):
+            def __init__(self, **kwargs):
+                super().__init__(**outer_kwargs, **kwargs)
+        return SpecificModel
+
+
+LogisticRegressionLasso = ModelFactory.get_model(LogisticRegression,
```
Comment (ElDeveloper) [inline q2_mlab/learningtask.py:67]: Just add a note about algorithm asymmetries

File: q2_mlab/tests/test_parameters.py [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+from sklearn.model_selection import ParameterGrid
+
+
+class Task(enum.Enum):
+    REGRESSION = 1
+    CLASSIFICATION = 2
+
+
+def create_synthetic_data(
+    n_samples, n_features, task=Task.REGRESSION, count=500
```
Comment (ElDeveloper) [inline q2_mlab/tests/test_parameters.py:20]: https://scikit-learn.org/stable/modules/generated/sklearn.datasets.make_classification.html

File: q2_mlab/tests/test_parameters.py [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+        # SVM
+        "LinearSVC",
+        "RadialSVC",
+        "SigmoidSVC",
+        # Other
+        "KNeighborsClassifier",
+        "MLPClassifier",
+    ]
+
+    def setUp(self):
```
Comment (ElDeveloper) [inline q2_mlab/tests/test_parameters.py:100]: Consider as a setUpClass method so that this gets only called once.

**Review by ElDeveloper — CHANGES_REQUESTED** [review]

File: q2_mlab/learningtask.py [hunk 6]
```diff
@@ -45,6 +47,21 @@
...
 
+class ModelFactory:
+    @staticmethod
+    def get_model(model_class, **outer_kwargs):
+        class SpecificModel(model_class):
+            def __init__(self, **kwargs):
+                super().__init__(**outer_kwargs, **kwargs)
+        return SpecificModel
+
+
```
Comment (gwarmstrong) [inline q2_mlab/learningtask.py:62]:
```suggestion

# scikit-learn has some asymmetries between classifiers and regressors, e.g.,
# `Lasso`'s classification equivalent is `LogisticRegression` with
# `penalty='l1'`. You can use ModelFactory.get_model to create Estimator
# classes that set specific parameters.
```

**Comment by ElDeveloper** [comment]
@patrickimran @gwarmstrong is this ready to be merged?

**Comment by gwarmstrong** [comment]
LGTM

**Pull request merged**